dot-net-sdk/validators/RuleValidator.cs

```diff
@@ -25,40 +26,73 @@ private static List<bool> EvaluateRuleCondition(SubjectAttributes subjectAttribu
 
     private static bool EvaluateCondition(SubjectAttributes subjectAttributes, Condition condition)
     {
-        if (subjectAttributes.ContainsKey(condition.attribute))
+        try
         {
-            subjectAttributes.TryGetValue(condition.attribute, out EppoValue outVal);
-            var value = outVal!;
-            Dictionary<OperatorType, Func<EppoValue, EppoValue, bool>> validationFunctions = new()
+            if (subjectAttributes.ContainsKey(condition.attribute) &&
+                subjectAttributes.TryGetValue(condition.attribute, out EppoValue outVal))
             {
-                { GTE, (a, b) => a.DoubleValue() >= b.DoubleValue() },
-                { GT, (a, b) => a.DoubleValue() > b.DoubleValue() },
-                { LTE, (a, b) => a.DoubleValue() <= b.DoubleValue() },
-                { LT, (a, b) => a.DoubleValue() < b.DoubleValue() },
+                var value = outVal!; // Assuming non-null for simplicity, handle nulls as necessary
+
+                if (condition.operatorType == GTE)
                 {
-                    MATCHES, (a, b) =>
-                        Regex.Match(a.StringValue(), b.StringValue(), RegexOptions.IgnoreCase).Success
-                },
+                    if (NuGet.Versioning.NuGetVersion.TryParse(value.StringValue(), out var valueSemver) &&
+                        NuGet.Versioning.NuGetVersion.TryParse(condition.value.StringValue(), out var conditionSemver))
+                    {
+                        return valueSemver >= conditionSemver;
+                    }
```

Comment:
attempt to coerce the incoming values into semver

<img width="703" alt="Screenshot 2024-02-16 at 11 10 08 AM" src="https://github.com/Eppo-exp/dot-net-server-sdk/assets/57361/6b46bde9-2158-4e0f-b8a9-1b9a5b1166ac">

otherwise fall-back to numeric evaluation